src/bin/set-version/set_version.rs

```diff
@@ -121,30 +121,85 @@ fn exec(args: VersionArgs) -> CargoResult<()> {
     if all {
         deprecated_message("The flag `--all` has been deprecated in favor of `--workspace`")?;
     }
-    let all = workspace || all;
-    let manifests = Manifests(resolve_manifests(
-        manifest_path.as_deref(),
-        all,
-        pkgid.as_deref().into_iter().collect::<Vec<_>>(),
-    )?);
 
     if dry_run {
         dry_run_message()?;
     }
 
     let workspace_members = workspace_members(manifest_path.as_deref())?;
 
-    exec_manifests(
-        manifests,
-        exclude,
-        target,
-        metadata,
-        dry_run,
-        workspace_members,
-    )
+    if workspace_version {
+        let workspace_manifest = LocalManifest::find(manifest_path.as_deref())?;
+        exec_workspace_version(
+            workspace_manifest,
+            target,
+            metadata,
+            dry_run,
+            workspace_members,
+        )
+    } else {
+        let all = workspace || all;
+        let manifests = Manifests(resolve_manifests(
+            manifest_path.as_deref(),
+            all,
+            pkgid.as_deref().into_iter().collect::<Vec<_>>(),
+        )?);
+
+        exec_manifests(
+            manifests,
+            exclude,
+            target,
+            metadata,
+            dry_run,
+            workspace_members,
+        )
+    }
+}
+
+/// Execute the command to modify `[workspace.package] version = "..."`.
+fn exec_workspace_version(
+    mut workspace_manifest: LocalManifest,
+    target: TargetVersion,
+    metadata: Option<String>,
+    dry_run: bool,
+    workspace_members: Vec<cargo_metadata::Package>,
+) -> CargoResult<()> {
+    let current = match workspace_manifest.get_workspace_version() {
+        Some(current) => current,
+        None => anyhow::bail!(
+            "'{}' does not have a `workspace.package.version` key, bumping/setting the workspace-wide version is not possible",
+            workspace_manifest.path.display()
+        ),
+    };
+
+    let next = match target.bump(&current, metadata.as_deref())? {
+        Some(next) => next,
+        None => anyhow::bail!("Failed to produce next version from current: {}", current),
+    };
+
+    workspace_manifest.set_workspace_version(&next);
+    upgrade_message("workspace-wide version", &current, &next)?;
+
+    let crate_root =
+        dunce::canonicalize(workspace_manifest.path.parent().expect("at least a parent"))?;
+
+    if let Some(ws_deps) = workspace_manifest.get_workspace_dependencies_mut() {
+        update_filtered_dependencies(
+            ws_deps,
```

Comment:
I'm not too confident in this updating the correct dependencies

Reply:
Adding a quick test with an unrelated dependency to the same version, you're correct, thanks for catching it, I'll fix

Reply:
Done, this was easier to fix than expected